sdk/go/common/resource/plugin/context_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/opentracing/opentracing-go/mocktracer"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/testing/diagtest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestContextRequest_race(t *testing.T) {
+	// Create a new mock tracer
+	tracer := mocktracer.New()
+
+	// Create a new root span
+	rootSpan := tracer.StartSpan("root")
+	ctx, err := NewContext(
+		diagtest.LogSink(t),
+		diagtest.LogSink(t),
+		nil,
+		nil,
+		t.TempDir(),
+		nil,
+		false,
+		rootSpan,
+	)
+	require.NoError(t, err)
+	// define waitgroup
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			ctx.Request()
+		}()
```

Comment:
Worth maybe adding a comment that says that we don't care about the result, this is just to exercise the data race.

Reply:
added descriptive comment above that should provide context.